vms/types/blob_data_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package types
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestJSON(t *testing.T) {
+	tests := []struct {
+		name         string
+		value        JSONByteSlice
+		expectedJSON string
+	}{
+		{
+			name:         "nil",
+			value:        nil,
+			expectedJSON: nullStr,
+		},
+		{
+			name:         "empty",
+			value:        []byte{},
+			expectedJSON: `"0x"`,
+		},
+		{
+			name:         "not empty",
+			value:        []byte{0, 1, 2, 3},
+			expectedJSON: `"0x00010203"`,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			require := require.New(t)
+
+			jsonBytes, err := json.Marshal(test.value)
+			require.NoError(err)
+			require.Equal(test.expectedJSON, string(jsonBytes))
+
+			var unmarshaled JSONByteSlice
```

Comment:
Does this need to support overwriting existing data (related to my comment about unmarshalling `"null"`)? If so, set this to something non-empty to begin with.

Reply:
I added an explicit test for the `null` behavior